app/components/post_body_additional_content/post_body_additional_content.js

```diff
@@ -87,16 +88,24 @@ export default class PostBodyAdditionalContent extends PureComponent {
         }
     }
 
-    load = (props) => {
+    load = async (props) => {
         const {link} = props;
         if (link) {
             let imageUrl;
             if (isImageLink(link)) {
                 imageUrl = link;
-            } else if (isYoutubeLink(link)) {
-                const videoId = getYouTubeVideoId(link);
-                imageUrl = `https://i.ytimg.com/vi/${videoId}/hqdefault.jpg`;
-                ImageCacheManager.cache(null, `https://i.ytimg.com/vi/${videoId}/default.jpg`, () => true);
+            }
+
+            if (!imageUrl) {
+                imageUrl = await getShortenedLink(link);
+                if (imageUrl) {
+                    this.setState({shortenedLink: imageUrl});
```

Comment:
Actually, this whole flow has me confused since we're setting imageUrl regardless of if it's an image at this point and there's some stuff confusing me about how we load the YouTube thumbnail. In my mind, this should look something like this
```
load = (props) => {
    const imageUrl = this.getImageUrl(props.link);
    if (imageUrl) {
        // cache image
        return;
    }

    getShortenedLink(props.link).then((shortened) => {
        const imageUrl = this.getImageUrl(props.link);
        if (imageUrl) {
            // cache image
        }
    });
});

getImageUrl = (link) => {
    let imageUrl;

    if (isImageLink(link) {
        imageUrl = link;
    } else if (isYouTubeLink(link)) {
        imageUrl = ...; // get thumnail
    }

    return imageUrl;
}
```
Alternatively, we have the getShortenedLink as part of getImageUrl and have getImageUrl always return a Promise, but that might just make it harder to follow